sycl/unittests/queue/Wait.cpp

```diff
@@ -120,37 +124,21 @@ TEST(QueueWait, QueueWaitTest) {
   // Blocked commands
   {
     TestContext = {};
-    buffer<int, 1> buf{range<1>(1)};
-
-    std::mutex m;
-    std::unique_lock<std::mutex> TestLock(m, std::defer_lock);
-    TestLock.lock();
-
-    event HostTaskEvent = Q.submit([&](handler &Cgh) {
-      auto acc = buf.template get_access<access::mode::read>(Cgh);
-      Cgh.host_task([=, &m]() {
-        (void)acc;
-        std::unique_lock<std::mutex> InsideHostTaskLock(m);
-      });
-    });
-    std::shared_ptr<detail::event_impl> HostTaskEventImpl =
-        detail::getSyclObjImpl(HostTaskEvent);
-    auto *Cmd = static_cast<detail::Command *>(HostTaskEventImpl->getCommand());
-    EXPECT_EQ(Cmd->MUsers.size(), 0u);
-    EXPECT_TRUE(Cmd->isHostTask());
-
-    // Use the host task to block the next commands
-    Q.submit([&](handler &Cgh) {
-      auto acc = buf.template get_access<access::mode::discard_write>(Cgh);
-      Cgh.fill(acc, 42);
-    });
-    Q.submit([&](handler &Cgh) {
-      auto acc = buf.template get_access<access::mode::discard_write>(Cgh);
-      Cgh.fill(acc, 42);
-    });
-    // Unblock the host task to allow the submitted events to complete once
-    // enqueued.
-    TestLock.unlock();
+    buffer<int, 1> Buf{range<1>(1)};
+
+    event DepEvent = submitTask(Q, Buf);
```

Comment:
Actually, since what this command does underneath is irrelevant, it should probably be replaced with a `MockCommand`.